### PR TITLE
Drop laminas-dom for symfony/dom-crawler

### PR DESCRIPTION
**Pull request by snapshotpl**
Signed-off-by: Witold Wasiczko <witold@wasiczko.pl>

|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | yes

Replaces `laminas/laminas-dom` (deprecated) to `symfony/dom-crawler`

### CODE REVIEW

File: src/PHPUnit/Controller/AbstractHttpControllerTestCase.php [hunk 1]
```diff
@@ -413,30 +413,28 @@ public function registerXpathNamespaces(array $xpathNamespaces)
...
+     * @return Crawler
      */
     private function query($path, $useXpath = false)
     {
         $response = $this->getResponse();
-        $document = new Document($response->getContent());
+        $document = new Crawler($response->getContent());
 
         if ($useXpath) {
-            $document->registerXpathNamespaces($this->xpathNamespaces);
```
Comment (snapshotpl) [inline src/PHPUnit/Controller/AbstractHttpControllerTestCase.php:433]: Not sure what used for. Not covered test to accept my changes. Anyone?

Comment (Ocramius) [reply src/PHPUnit/Controller/AbstractHttpControllerTestCase.php:433]: It basically sets the scope for XML search operations: without this, some elements would not necessarily be discoverable OOTB.

Comment (snapshotpl) [reply src/PHPUnit/Controller/AbstractHttpControllerTestCase.php:433]: I have added test case but I cannot make it pass. Is test or implementation wrong?

Comment (snapshotpl) [reply src/PHPUnit/Controller/AbstractHttpControllerTestCase.php:433]: Copied example from Symfony webpage works!

**Comment by snapshotpl** [comment]
Think about change base to 3.6 from 4.0.

**Comment by Ocramius** [comment]
@snapshotpl I can't see any API changes: this looks fine for `3.x`, IMO?

**Comment by snapshotpl** [comment]
@Ocramius now better :)

File: src/PHPUnit/Controller/AbstractHttpControllerTestCase.php [hunk 2]
```diff
@@ -422,30 +422,28 @@ public function registerXpathNamespaces(array $xpathNamespaces)
      *
      * @param  string $path
      * @param  bool $useXpath
-     * @return Document\NodeList
+     * @return Crawler
```
Comment (froschdesign) [inline src/PHPUnit/Controller/AbstractHttpControllerTestCase.php:425]: @Ocramius 
This return type was changed.

Comment (snapshotpl) [reply src/PHPUnit/Controller/AbstractHttpControllerTestCase.php:425]: Sure, but it's private

Comment (froschdesign) [reply src/PHPUnit/Controller/AbstractHttpControllerTestCase.php:425]: I am blind, because I was sure I had read `public`.

**Review by froschdesign — APPROVED** [review]

**Review by Ocramius — APPROVED** [review]
LGTM :+1:

**Comment by Ocramius** [comment]
@snapshotpl are you able to perhaps dis-entangle #45? :D

**Comment by snapshotpl** [comment]
Sure. laminas-console is removed in 4.x

**Pull request merged**